src/main.rs

```diff
@@ -23,15 +24,42 @@ const PLATFORM: &'static [PlatformData] = &[
 fn main() {
     let mut col = Collector::new();
 
+	let matches = App::new("conga")
+        .version("0.1.0")
+        .author("")
+        .about("System Configurator Gatherer")
+        .arg(Arg::with_name("verbose")
+            .short("v") // TODO: currently unused: set up logging first
+            .long("verbose")
+            .help("increase verbosity")
+            .takes_value(false))
+        .arg(Arg::with_name("allow-failures")
+            .short("-f")
+            .long("allow-failures")
+            .help("don't abort if a collector fails")
+            .takes_value(false))
+        .get_matches();
+
     // Iterate through each tag:func pair
     //   Call the function, and insert the data from the collecting function
-    // TODO: add error checking here?
     for pd in PLATFORM {
         let f = pd.func;
-        // YOLO: actually check the error instead of just unwrap >:(
-        let data = f(&mut col).unwrap();
-        // TODO: Perhaps error check here
-        col.add_data(String::from(pd.tag), data)
+
+        let data = f(&mut col);
+
+        if let Ok(d) = data {
+            col.add_data(String::from(pd.tag), d)
+        }
+        else if matches.is_present("allow-failures") {
+            // TODO: Perhaps instead of dropping the tag, setting to either the
+            //   error result, or some kind of "null" value.
```

Comment:
Change `TODO` to instead record the error in some new defined type. it should be obvious to a parser that this is an error and not error text coming from the run command.

For now, record a `null` or something here. Shouldn't silently drop tags.